pyogrio/tests/test_path.py

```diff
@@ -0,0 +1,135 @@
+        # cloud URIs
+        ("s3://testing/data.gpkg", "/vsis3/testing/data.gpkg"),
+        ("gs://testing/data.gpkg", "/vsigs/testing/data.gpkg"),
+        ("az://testing/data.gpkg", "/vsiaz/testing/data.gpkg"),
+        ("adl://testing/data.gpkg", "/vsiadls/testing/data.gpkg"),
+        ("adls://testing/data.gpkg", "/vsiadls/testing/data.gpkg"),
+        # archives
+        ("zip://data.zip", "/vsizip/data.zip"),
+        ("tar://data.tar", "/vsitar/data.tar"),
+        ("gzip://data.gz", "/vsigzip/data.gz"),
+        ("tar://./my.tar!my.geojson", "/vsitar/./my.tar/my.geojson"),
+        (
+            "zip://home/data/shapefile.zip!layer.shp",
+            "/vsizip/home/data/shapefile.zip/layer.shp",
+        ),
+        # combined schemes
+        ("zip+s3://testing/shapefile.zip", "/vsizip/vsis3/testing/shapefile.zip"),
+        (
+            "zip+https://s3.amazonaws.com/testing/shapefile.zip",
+            "/vsizip/vsicurl/https://s3.amazonaws.com/testing/shapefile.zip",
+        ),
+    ],
+)
+def test_vsi_path(path, expected):
+    assert vsi_path(path) == expected
+
+
+def test_vsi_handling_read_functions(naturalearth_lowres_vsi):
+    # test that all different read entry points have the path handling
+    # (a zip:// path would otherwise fail)
+    path, _ = naturalearth_lowres_vsi
+    path = "zip://" + str(path)
+
+    result = pyogrio.raw.read(path)
+    assert len(result[1]) == 177
+
+    result = pyogrio.read_info(path)
+    assert result["features"] == 177
+
+    result = pyogrio.read_bounds(path)
+    assert len(result[0]) == 177
+
+    result = pyogrio.read_dataframe(path)
+    assert len(result) == 177
+
+
+def test_path_absolute(data_dir):
+    # pathlib path
+    path = data_dir / "naturalearth_lowres/naturalearth_lowres.shp"
+    df = pyogrio.read_dataframe(path)
+    len(df) == 177
+
+    # str path
+    df = pyogrio.read_dataframe(str(path))
+    len(df) == 177
+
+
+def test_path_relative(data_dir):
+    with change_cwd(data_dir):
+        df = pyogrio.read_dataframe("naturalearth_lowres/naturalearth_lowres.shp")
+    len(df) == 177
+
+
+def test_uri_local_file(data_dir):
+    uri = "file://" + str(data_dir / "naturalearth_lowres/naturalearth_lowres.shp")
+    df = pyogrio.read_dataframe(uri)
+    len(df) == 177
+
+
+def test_zip_path(naturalearth_lowres_vsi):
+    path, path_vsi = naturalearth_lowres_vsi
+    path_zip = "zip://" + str(path)
+
+    # absolute zip path
+    df = pyogrio.read_dataframe(path_zip)
+    assert len(df) == 177
+
+    # relative zip path
+    with change_cwd(path.parent):
+        df = pyogrio.read_dataframe("zip://" + path.name)
+    assert len(df) == 177
+
+    # absolute vsizip path
+    df = pyogrio.read_dataframe(path_vsi)
+    assert len(df) == 177
+
+
+@pytest.mark.network
+def test_url():
+    df = pyogrio.read_dataframe(
+        "https://raw.githubusercontent.com/geopandas/pyogrio/main/pyogrio/tests/fixtures/naturalearth_lowres/naturalearth_lowres.shp"
```

Comment:
Are we OK with having tests that require network? 
I added a marker for now, so at least it is easy to skip them (but by default this marker doesn't do anything)

Reply:
Yes, I think it is fine. It is surely useful and we have similar one in geopandas as well.